openquake/hazardlib/geo/surface/simple_fault.py

```diff
@@ -171,17 +172,22 @@ def from_fault_data(cls, fault_trace, upper_seismogenic_depth,
 
     @classmethod
     def get_fault_vertices_3d(cls, fault_trace, upper_seismogenic_depth,
-                              lower_seismogenic_depth, dip):
+                              lower_seismogenic_depth, dip, index_patch=None):
         """
-        Get surface main vertexes.
-
+        Get surface main vertices.
         Parameters are the same as for :meth:`from_fault_data`, excluding
         mesh spacing.
 
+        :param index_patch:
+            indicate the patch of the fault to output the vertices.
+            The first patch strating from where the fault trace starts.
         :returns:
```

Comment:
If the case when the index patch is not needed you can delete it. Otherwise the more general option is to return three np.vectors (lon, lat, deps) or a single n x 3 numpy array. 

I think it better if you fix this before I continue reviewing the following pull request.